code/modules/spells/spell_types/genetic.dm

```diff
@@ -28,7 +28,8 @@
 		for(var/A in traits)
 			ADD_TRAIT(target, A, GENETICS_SPELL)
 		active_on += target
-		addtimer(CALLBACK(src, .proc/remove, target), duration)
+		if(duration < charge_max)

```

Comment:
This doesn't make a lot of sense imo. Just make this a real toggle if the spell is upgraded enough.

Reply:
This makes it so that if an admin lowers the spell's duration or if we add some other temporary self-augging spells that have less than 4 levels in the future (like, say, a Haste spell), this feature will accommodate them.

Actually, that Haste spell is a pretty cool idea. I might try to create it (and/or an Expeditious Retreat spell) in the future...